src/foam/core/AbstractFObject.java

```diff
@@ -82,4 +82,11 @@ public boolean hasOwnProperty(String prop) {
     PropertyInfo property = (PropertyInfo) getClassInfo().getAxiomByName(prop);
     return property != null && property.hasOwnProperty(this);
   }
+
+  public boolean hasDefaultValue(String prop) {
+    if ( ! this.hasOwnProperty(prop) )
```

Comment:
BTW: hasOwnProperty() is a Javascript thing. Seems odd having it for Java. Would be better if called isSet() or isPropertySet() or something like that.

Reply:
Changed the function names